addons/popochiu/editor/main_dock/popochiu_dock.gd

```diff
@@ -341,4 +305,26 @@ func _check_node(node: Node) -> void:
 		node.set_name.call_deferred("Character%s *" % node.script_name)
 
 
+func _get_popochiu_objects_resources(dir_name: String, type_key: Constants.Types) -> Resource:
+	var resource_filesystem := EditorInterface.get_resource_filesystem()
+	var dir_path: String = _types[type_key].path + dir_name
+	
+	for file_name: String in DirAccess.get_files_at(dir_path):
+		if file_name.get_extension() != "tres": continue
+		
+		var resource: Resource = load(dir_path.path_join(file_name))
+		if not (
```

Comment:
May also be this? Not sure it's better, maybe just more readable... but your version is more robust because it is not based on strings.

You decide.

```suggestion
		if not (resource.get_class() in ["PopochiuRoomData", "PopochiuCharacterData", "PopochiuInventoryItemData", "PopochiuDialog"]:
```

Reply:
That won't work because the `get_class()` method in this case will return "Resource". In Godot, the classes declared with `class_name` are not listed as classes. WTF with this engine? I don't what I love it so much...

Reply:
Ah, right!

We love it because it's free as in free speech, lightweight and open to be understood. But this doesn't make it perfect :)

Anyway, you are right, I'm closing this comment. Thanks!